src/term/mod.rs

```diff
@@ -258,6 +267,10 @@ impl Book {
     let name = self.def_names.remove(def_id);
     name.zip(def)
   }
+
+  pub fn is_generated_def(&self, def_id: DefId) -> bool {
```

Comment:
This causes confusion with the Definition.generated field.
Here we're talking about a generated name. There it's a generated body.
Could you add this distinction in both places?